src/core/types.rs

```diff
@@ -100,6 +100,26 @@ impl Type {
             two_1, two_2, two_4, two_8, two_16, two_32, two_64, two_128, two_256,
         ]
     }
+
+    /// Return an array `pow2s` of types such that `pow2s[i] = 2^i` holds for 0 ≤ `i` < n.
+    pub fn powers_of_two_vec(n: usize) -> Vec<Arc<Type>> {
+        assert!(n < 16);
```

Comment:
In 4838302e05e8b1b0b4ef41228cdc96cedb57e1d8:

We could probably drop this assertion. But if we keep it, we should put it in the doccomment.